analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -326,15 +326,15 @@ private object ReadNumberValue()
             object BuildResult()
```

Comment:
I would refactor this method to use `BigInteger` and `long` for integral, and `double` for decimals, as follows:
```cs
object BuildResult()
{
    if (@decimal == null)
    {
        var integralValue = BigInteger.Parse(integer.ToString());
        if (exponent != null)
        {
            if (exponent.Length == 0 || exponent.ToString() == "-")
            {
                throw new JsonException($"Unexpected number exponent format: {exponent}", LastStart);
            }
            else
            {
                integralValue *= BigInteger.Pow(10, int.Parse(exponent.ToString()));
            }
        }

        return integralValue < long.MinValue || integralValue > long.MaxValue ? integralValue : (long)integralValue;
    }
    else
    {
        var decimalValue = double.Parse(integer + CultureInfo.CurrentCulture.NumberFormat.NumberDecimalSeparator + @decimal);
        if (exponent != null)
        {
            if (exponent.Length == 0 || exponent.ToString() == "-")
            {
                throw new JsonException($"Unexpected number exponent format: {exponent}", LastStart);
            }
            else
            {
                decimalValue *= Math.Pow(10, int.Parse(exponent.ToString()));
            }
        }
        return decimalValue;
    }
}
```

To be thoroughly tested.

Unfortunately `System.Numerics` lacks of a `BigRational` type, that is still under discussion: https://github.com/dotnet/runtime/issues/71791

Reply:
I did the change - parsing a big integer instead of a double-  and there are lots of tests in different rules/other pieces of code that are affected.

Although I don't like at all "ephemeral" fixes - that usually end up being long-lived ones, in this case, this change requires its own PR and time.
I currently don't have the second as it's out-of-sprint work.

What we have now it's still better than what was there IMO - I'd definitely open a hardening issue but not sure about the priority.

Reply:
I tried to apply this change that seems to be a very good improvement (I think we should open a hardening ticket to follow up),
but it's more work than expected in the context of out of sprint PR.

2 options from my side:
- Either I close this PR  - write down in the ticket what needs to be done and move on (but it's still an AD0001- however not sure how urgent it is as this code has been there for quite some time). 
- We merge what we have now - and open a follow-up issue to update the BuildResult method.

Both are fine by me :) 

WDYT?

Reply:
I think the first option is the most pragmatic: let's go in baby steps and fix this AD0001 first.
I agree we should have a ticket to improve this in some of the upcoming hardening sprints.

Reply:
I'll open an issue later today.

Reply:
@antonioaversa  Feel free to edit as you see fit https://github.com/SonarSource/sonar-dotnet/issues/8344